source/particle/property/interface.cc

```diff
@@ -215,21 +215,60 @@ namespace aspect
         Vector<double> solution;
         std::vector<Tensor<1,dim>> gradient;
 
+        bool need_solution = false;
+        bool need_gradient = false;
+        unsigned int n_components = numbers::invalid_unsigned_int;
+
+        if (inputs.solution.size() > 0)
+          {
+            n_components = inputs.solution[0].size();
+
+            for (unsigned int i=0; i<n_components; ++i)
+              if (get_update_flags(i) & update_values)
+                {
+                  need_solution = true;
+                  break;
+                }
+          }
+
+        if (inputs.gradients.size() > 0)
+          {
+            n_components = inputs.gradients[0].size();
+
+            for (unsigned int i=0; i<n_components; ++i)
+              if (get_update_flags(i) & update_gradients)
+                {
+                  need_gradient = true;
+                  break;
+                }
+          }
+
         unsigned int i = 0;
         for (auto particle = particles.begin(); particle != particles.end(); ++particle)
           {
-            if (get_needed_update_flags() & update_values)
+            if (need_solution)
               {
                 solution.reinit(inputs.solution[i].size());
                 for (unsigned int j=0; j<solution.size(); ++j)
-                  solution[j] = inputs.solution[i][j];
+                  {
+                    if (get_update_flags(j) & update_values)
+                      solution[j] = inputs.solution[i][j];
+                    else
+                      solution[j] = numbers::signaling_nan<double>();
```

Comment:
Should we do this only in debug mode for performance reasons or do you think this doesn't matter?